### DISCUSSION TITLE
Release Notes with Shared Query

### DISCUSSION THREAD

**Discussion by keepcalmandbuild** [rfennell/AzurePipelines#1475] [Ideas]
Hello, I love the GenerateReleaseNotes Node based Cross Platform extension and I am exploring how to capture a baseline of known bugs at the time of the release. I read about wiqlFromTarget and wiqlWhereClause. I would like a similar functionality, that uses an existing shared query ID instead of wiql to specify the work item query for release notes. My motivation is that I would like to be able to have only one query, to be sure that data visualization is consistent across different "media". For example, I would like to reuse the same query for release notes and dashboard.
I have thought of including a query link in the markdown template and publish it in the wiki as a workaround, but in that case the wiki would show the current results of the query at the time of opening the wiki, not the results at the time of release notes generation.

**Comment by rfennell** [comment]
The only reason I could see this not working would be permissions, if the agent running the build did not have access to read shared queries. Even if this was the case it should only be a case of altering permissions.

The basic API call (via the Node SDK) is along the lines of 

`https://dev.azure.com/<org>/<project>/_apis/wit/queries/Shared%20Queries/Query1?$expand=all&api-version=7.0`

From this call we can get the JSON for the WIQL

There would need to be some string handling to extract the where clause of the WIQL (as the task handles the columns used), but all doable

I will promote this discussion to be a tracked enhancement, not sure when I will get around to it

**Comment by rfennell** [comment]
Of course I am happy to consider PRs

**Comment by arulconsultant** [comment]
In my case, have tried utilizing shared queries by specifying 
::: query-table <<queryidentifier>>

have extracted queryid from the basic API call using above guidance from @rfennell
Anything i can do to pass that value as input to the inline template?

            # Associated Workitems
            | ID | Work Item Type | Title | Assigned To | State | Tags |
            |-|-|-|-|-|-|
            {{#forEach this.workItems}}
            |{{this.id}} | {{lookup this.fields 'System.WorkItemType'}} | {{lookup this.fields 'System.Title'}} | {{#with (lookup this.fields 'System.AssignedTo')}} {{displayName}} {{/with}} | {{lookup this.fields 'System.State'}} | {{lookup this.fields 'System.Tags'}} |
            {{/forEach}}
            ::: query-table '{{queryidentifier}}'

**Reply by keepcalmandbuild** [reply]
When you use a query-table in the template as in your example, I think it will execute the query at the time when a user opens that wiki page, so data may be different every time. I initiated this discussion because I needed the query to be executed once, so I could write the baselined data in the file. In my case, I formatted the queryWorkItems object observing its structure in the raw json file produced by the tool.
If I remember correctly, I used the path and name of the query as a parameter of the tool to execute the query, not the id of the query.
